### ISSUE TITLE
RedirectTo URL no longer works for PKCE SSR use cases 

### ISSUE THREAD

**Issue by djshubs** [supabase/gotrue#1271]
# Bug report

- [ X] I confirm this is a bug with Supabase, not with my own application. 
- [X ] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug
In the `auth/callback` code, I use the 'next' searchParam to redirect my user after the code exchange has occurred. I accomplish this by using `emailRedirectTo with `next` searchParams when I call [supabse.auth.signInWithOtp](https://supabase.com/docs/reference/javascript/auth-signinwithoauth). 

However, with `[auth/confirm](https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr#create-api-endpoint-for-handling-tokenhash)` approach, despite having the `next` searchParams in the code, I can't use the emailRedirectTo at all. This is because the email templates only take .TokenHash and .SiteURL as params and I would have to hard code the next param (see [Invite user Template](https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr#update-email-templates-with-url-for-api-endpoint)).

With the .ConfirmationURL, it would append the auth/callback?next=/location as a redirect_to parameter. However, that's not possible if I update the templates for PKCE flow with SSR. 

tl;dr - .ConfirmationURL supports redirectTo in the url allowing me to redirect users after `auth/callback`. Using `auth/confirm` in email templates, prevents me from using `redirectTo` as a result I can't dynamically route users post authentication.

## To Reproduce

Using emailRedirectTo in signInWithOtp will not redirect user if using SSR and SiteURL email templates with Token Hash appended to Auth/confirm url. 

## Expected behavior

The expectation is that previous capability of emailRedirectTo works with SSR. 

## Additional context
Our use case is that when we have a user that can be invited to be an instructor across many of our customers. So when they are invited, the next url links them to a customer specific page.

We are able to do that with non-SSR cases using the redirect to param.

We have another situation where a user need to authenticate themselves before checkout can occur. We use the redirect to param, today, to forward them to the next link.
